app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,47 @@
+class PasswordResetsController < ApplicationController
+  skip_before_filter :require_login
+
+  # request password reset.
+  # you get here when the user entered his email in the reset password form and submitted it.
+  def create 
+    @user = User.find_by_email(params[:email])
+
+    # This line sends an email to the user with instructions on how to reset their password (a url with a random token)
+    @user.deliver_reset_password_instructions! if @user
+    #puts @user.email
+    # Tell the user instructions have been sent whether or not email was found.
+    # This is to not leak information to attackers about which emails exist in the system.
+    redirect_to(root_path, :notice => 'Instructions have been sent to your email.')
```

Comment:
Though i have removed the commenst.. but we can keep them for better redability. since when we come back to see it,we know what is happening there just by reading it, atleast for new users.